jdk: 15.0.1-ga -> 16+36

petabyteboy opened the pull request:
<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!
List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

Built jdk16 on x86_64-linux and i686-linux.
The version string doesn't look quite right yet:

```
OpenJDK Runtime Environment (build 16+0-adhoc..source)
```

###### Motivation for this change

> JDK 16 reached General Availability on 16 March 2021

https://openjdk.java.net/projects/jdk/16/


###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

Commit:
jdk: 15.0.1-ga -> 16+36

SuperSandro2000 commented on pkgs/development/compilers/openjdk/15.nix:
```diff
@@ -0,0 +1,158 @@
+      mv build/*/images/jdk $out/lib/openjdk
+
+      # Remove some broken manpages.
+      rm -rf $out/lib/openjdk/man/ja*
```
```suggestion
      rm -r $out/lib/openjdk/man/ja*
```
just to make sure we remove something. Maybe they move in the future or rename, etc.

SuperSandro2000 commented on pkgs/development/compilers/openjdk/15.nix:
```diff
@@ -0,0 +1,158 @@
+      ln -s $out/include/linux/*_md.h $out/include/
+
+      # Remove crap from the installation.
+      rm -rf $out/lib/openjdk/demo
```
```suggestion
      rm -r $out/lib/openjdk/demo
```

SuperSandro2000 commented on pkgs/development/compilers/openjdk/15.nix:
```diff
@@ -0,0 +1,158 @@
+      license = licenses.gpl2;
+      description = "The open-source Java Development Kit";
+      maintainers = with maintainers; [ edwtjo ];
+      platforms = [ "i686-linux" "x86_64-linux" "aarch64-linux" "armv7l-linux" "armv6l-linux" ];
```
Maybe simply only linux instead of 5 arches of it?

petabyteboy replied:
I think the bootstrap jdk is only available for those architectures.

petabyteboy replied:
By the way all your comments are on a file that was not actually changed. Feel free to make a PR to improve the jdk{11,13,14,15} packages, but this is not the topic of this PR.

SuperSandro2000 commented:
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).
If you have any questions or problems please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 116717` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
18 packages marked as broken and skipped:
</summary>
<ul>
<li>aldor</li>
<li>foo-yc20</li>
<li>jboss</li>
<li>junit</li>
<li>junixsocket</li>
<li>octave-jit</li>
<li>octavePackages.fem-fenics</li>
<li>octavePackages.vibes</li>
<li>octavePackages.vrml</li>
<li>openfire</li>
<li>pig</li>
<li>plm</li>
<li>purePackages.faust</li>
<li>python38Packages.skein</li>
<li>python39Packages.skein</li>
<li>spark</li>
<li>spring</li>
<li>springLobby</li>
</ul>
</details>
<details>

  <summary>5 packages failed to build and already failed to build on hydra master:</summary>
  <ul>
<li>graalvm8: log was empty</li>
<li>jvmci8: log was empty</li>
<li>magnetophonDSP.MBdistortion: log was empty</li>
<li>sweethome3d.furniture-editor: log was empty</li>
<li>sweethome3d.textures-editor: log was empty</li>
  </ul>
</details>
<details>

  <summary>18 packages failed to build and are new build failures:</summary>
  <ul>
<li>kodi: log was empty</li>
<li>(kodiPackages.kodi): log was empty</li>
<li>kodi-gbm: log was empty</li>
<li>kodi-wayland: log was empty</li>
<li>kodiPackages.inputstream-adaptive: log was empty</li>
<li>kodiPackages.joystick: log was empty</li>
<li>kodiPackages.kodi-platform: log was empty</li>
<li>kodiPackages.pvr-hdhomerun: log was empty</li>
<li>kodiPackages.pvr-hts: log was empty</li>
<li>kodiPackages.pvr-iptvsimple: log was empty</li>
<li>kodiPackages.steam-controller: log was empty</li>
<li>kodiPackages.vfs-libarchive: log was empty</li>
<li>kodiPackages.vfs-sftp: log was empty</li>
<li>mindustry: log was empty</li>
<li>mindustry-server: log was empty</li>
<li>mindustry-wayland: log was empty</li>
<li>plantuml-server: log was empty</li>
<li>yed: log was empty</li>
  </ul>
</details>
<details>
<summary>
313 packages built:
</summary>
<ul>
<li>alchemy</li>
<li>alda</li>
<li>ammonite (ammonite_2_13)</li>
<li>ammonite_2_12</li>
<li>apache-directory-studio</li>
<li>apache-flex-sdk</li>
<li>apktool</li>
<li>archi</li>
<li>arduino</li>
<li>arduino-core</li>
<li>asciidoc-full-with-plugins</li>
<li>asciidoctorj</li>
<li>aspectj</li>
<li>astrolabe-generator</li>
<li>atlassian-cli</li>
<li>avro-tools</li>
<li>aws_mturk_clt</li>
<li>awstats</li>
<li>axis2</li>
<li>basex</li>
<li>bfg-repo-cleaner</li>
<li>bftools</li>
<li>bitwig-studio1</li>
<li>bitwig-studio2</li>
<li>bloop</li>
<li>bonnmotion</li>
<li>boot</li>
<li>briss</li>
<li>ccemux</li>
<li>cfr</li>
<li>checkstyle</li>
<li>clooj</li>
<li>closurecompiler</li>
<li>confluent-platform</li>
<li>coursier</li>
<li>cryptomator</li>
<li>davmail</li>
<li>dex2jar</li>
<li>ditaa</li>
<li>domination</li>
<li>ec2_api_tools</li>
<li>emem</li>
<li>emscripten</li>
<li>epubcheck</li>
<li>ergo</li>
<li>exhibitor</li>
<li>faust (faust2)</li>
<li>faust2alqt</li>
<li>faust2alsa</li>
<li>faust2csound</li>
<li>faust2firefox</li>
<li>faust2jack</li>
<li>faust2jackrust</li>
<li>faust2jaqt</li>
<li>faust2ladspa</li>
<li>faust2lv2</li>
<li>faustlive</li>
<li>flexibee</li>
<li>flink</li>
<li>flyway</li>
<li>frostwire</li>
<li>frostwire-bin</li>
<li>gama</li>
<li>ganttproject-bin</li>
<li>geoipjava</li>
<li>gephi</li>
<li>global-platform-pro</li>
<li>gogui</li>
<li>gpsprune</li>
<li>graylog</li>
<li>gremlin-console</li>
<li>groove</li>
<li>groovy</li>
<li>h2</li>
<li>haxePackages.hxjava</li>
<li>hdfview</li>
<li>hsqldb</li>
<li>html5validator</li>
<li>hydraAntLogger</li>
<li>imagej</li>
<li>ino</li>
<li>ipscan</li>
<li>jadx</li>
<li>jameica</li>
<li>jbake</li>
<li>jbang</li>
<li>jd-gui</li>
<li>jdiskreport</li>
<li>jdk (jdk16 ,jre ,openjdk ,openjdk16)</li>
<li>jre_headless (jdk16_headless ,openjdk16_headless ,openjdk_headless)</li>
<li>jedit</li>
<li>jflex</li>
<li>jicofo</li>
<li>jing (jing-trang)</li>
<li>jitsi-videobridge</li>
<li>jmeter</li>
<li>jnetmap</li>
<li>josm</li>
<li>jre_minimal</li>
<li>jruby</li>
<li>jugglinglab</li>
<li>jython</li>
<li>kapitonov-plugins-pack</li>
<li>key</li>
<li>keycloak</li>
<li>kotlin</li>
<li>ktlint</li>
<li>languagetool</li>
<li>leiningen</li>
<li>leo3-bin</li>
<li>libreoffice-fresh</li>
<li>libreoffice-fresh-unwrapped</li>
<li>libreoffice-qt</li>
<li>libreoffice-still</li>
<li>libreoffice-unwrapped (libreoffice-still-unwrapped)</li>
<li>librsb</li>
<li>liquibase</li>
<li>logisim</li>
<li>logstash (logstash6)</li>
<li>logstash6-oss</li>
<li>logstash7</li>
<li>logstash7-oss</li>
<li>lombok</li>
<li>ma1sd</li>
<li>macse</li>
<li>magnetophonDSP.CharacterCompressor</li>
<li>magnetophonDSP.CompBus</li>
<li>magnetophonDSP.ConstantDetuneChorus</li>
<li>magnetophonDSP.LazyLimiter</li>
<li>magnetophonDSP.RhythmDelay</li>
<li>magnetophonDSP.VoiceOfFaust</li>
<li>magnetophonDSP.faustCompressors</li>
<li>magnetophonDSP.pluginUtils</li>
<li>magnetophonDSP.shelfMultiBand</li>
<li>makemkv</li>
<li>martyr</li>
<li>marvin</li>
<li>maven (maven3)</li>
<li>mediathekview</li>
<li>metals</li>
<li>micronaut</li>
<li>mill</li>
<li>minc_widgets</li>
<li>minecraft</li>
<li>minecraft-server</li>
<li>mkgmap</li>
<li>mkgmap-splitter</li>
<li>mooSpace</li>
<li>msgviewer</li>
<li>munin</li>
<li>mxisd</li>
<li>nailgun</li>
<li>netlogo</li>
<li>nexus</li>
<li>nzbhydra2</li>
<li>octave</li>
<li>octaveFull</li>
<li>octavePackages.arduino</li>
<li>octavePackages.audio</li>
<li>octavePackages.bim</li>
<li>octavePackages.bsltl</li>
<li>octavePackages.cgi</li>
<li>octavePackages.communications</li>
<li>octavePackages.control</li>
<li>octavePackages.data-smoothing</li>
<li>octavePackages.database</li>
<li>octavePackages.dataframe</li>
<li>octavePackages.dicom</li>
<li>octavePackages.divand</li>
<li>octavePackages.doctest</li>
<li>octavePackages.econometrics</li>
<li>octavePackages.financial</li>
<li>octavePackages.fits</li>
<li>octavePackages.fpl</li>
<li>octavePackages.fuzzy-logic-toolkit</li>
<li>octavePackages.ga</li>
<li>octavePackages.general</li>
<li>octavePackages.generate_html</li>
<li>octavePackages.geometry</li>
<li>octavePackages.gsl</li>
<li>octavePackages.image</li>
<li>octavePackages.image-acquisition</li>
<li>octavePackages.instrument-control</li>
<li>octavePackages.interval</li>
<li>octavePackages.io</li>
<li>octavePackages.level-set</li>
<li>octavePackages.linear-algebra</li>
<li>octavePackages.lssa</li>
<li>octavePackages.ltfat</li>
<li>octavePackages.mapping</li>
<li>octavePackages.matgeom</li>
<li>octavePackages.miscellaneous</li>
<li>octavePackages.msh</li>
<li>octavePackages.mvn</li>
<li>octavePackages.nan</li>
<li>octavePackages.ncarray</li>
<li>octavePackages.netcdf</li>
<li>octavePackages.nurbs</li>
<li>octavePackages.ocl</li>
<li>octavePackages.octclip</li>
<li>octavePackages.octproj</li>
<li>octavePackages.optics</li>
<li>octavePackages.optim</li>
<li>octavePackages.optiminterp</li>
<li>octavePackages.parallel</li>
<li>octavePackages.quaternion</li>
<li>octavePackages.queueing</li>
<li>octavePackages.signal</li>
<li>octavePackages.sockets</li>
<li>octavePackages.sparsersb</li>
<li>octavePackages.splines</li>
<li>octavePackages.statistics</li>
<li>octavePackages.stk</li>
<li>octavePackages.strings</li>
<li>octavePackages.struct</li>
<li>octavePackages.symbolic</li>
<li>octavePackages.tisean</li>
<li>octavePackages.tsa</li>
<li>octavePackages.video</li>
<li>octavePackages.windows</li>
<li>octavePackages.zeromq</li>
<li>odpdown</li>
<li>omegat</li>
<li>opa</li>
<li>open-ecard</li>
<li>openapi-generator-cli</li>
<li>openapi-generator-cli-unstable</li>
<li>openems</li>
<li>opengrok</li>
<li>openjdk16-bootstrap</li>
<li>openjump</li>
<li>papermc</li>
<li>perl530Packages.NetZooKeeper</li>
<li>perl532Packages.NetZooKeeper</li>
<li>picard-tools</li>
<li>picolisp</li>
<li>plantuml</li>
<li>portmod</li>
<li>prometheus-jmx-httpserver</li>
<li>python38Packages.databricks-connect</li>
<li>python38Packages.python-csxcad</li>
<li>python38Packages.python-openems</li>
<li>python38Packages.sphinxcontrib_plantuml</li>
<li>python39Packages.databricks-connect</li>
<li>python39Packages.python-csxcad</li>
<li>python39Packages.python-openems</li>
<li>python39Packages.sphinxcontrib_plantuml</li>
<li>quantomatic</li>
<li>rascal</li>
<li>riemann</li>
<li>runelite</li>
<li>saxon</li>
<li>saxon-he</li>
<li>saxonb (saxonb_8_8)</li>
<li>saxonb_9_1</li>
<li>sbt (simpleBuildTool)</li>
<li>sbt-extras</li>
<li>sbt-with-scala-native</li>
<li>scalafmt</li>
<li>schemaspy</li>
<li>selendroid</li>
<li>selenium-server-standalone</li>
<li>seyren</li>
<li>shattered-pixel-dungeon</li>
<li>smc</li>
<li>snpeff</li>
<li>soapui</li>
<li>solr</li>
<li>sonar-scanner-cli</li>
<li>sourcetrail</li>
<li>spring-boot-cli</li>
<li>squirrel-sql</li>
<li>sshoogr</li>
<li>stm32cubemx</li>
<li>storm</li>
<li>subgit</li>
<li>subsonic</li>
<li>swagger-codegen</li>
<li>sweethome3d.application</li>
<li>swingsane</li>
<li>swt</li>
<li>tabula</li>
<li>tabula-java</li>
<li>tambura</li>
<li>teensyduino</li>
<li>tessera</li>
<li>tomcat-native</li>
<li>tomcat_connectors</li>
<li>tr-patcher</li>
<li>turbovnc</li>
<li>tuxguitar</li>
<li>tvbrowser-bin</li>
<li>umlet</li>
<li>unciv</li>
<li>unoconv</li>
<li>varscan</li>
<li>vassal</li>
<li>vimPlugins.lens-vim</li>
<li>vimPlugins.vim-grammarous</li>
<li>visualvm</li>
<li>vue</li>
<li>vuze</li>
<li>weka</li>
<li>workcraft</li>
<li>xmind</li>
<li>xmloscopy</li>
<li>youtrack</li>
<li>yuicompressor</li>
<li>zap</li>
<li>zipkin</li>
<li>zkfuse</li>
<li>zookeeper_mt</li>
<li>zxing</li>
</ul>
</details>

roberth commented:
Please do not remove JDK major versions. Projects tend to be picky about those.

roberth commented:
Seems like a good idea to call it `16.nix` instead of `default.nix`. That will save renaming when 17 comes out.

roberth commented:
I've overestimated upstream's support window. 15 should be marked insecure.

jerith666 commented:
IMO we should not carry insecure, unsupported JDK versions in a rolling-release distro like NixOS / nixpkgs; we should carry the current LTS release and the current release, nothing more.  If users need an intermediate release like 15, they should pin nixpkgs at an older revisions where it was present.

SuperSandro2000 commented:
> IMO we should not carry insecure, unsupported JDK versions in a rolling-release distro like NixOS / nixpkgs; we should carry the current LTS release and the current release, nothing more. If users need an intermediate release like 15, they should pin nixpkgs at an older revisions where it was present.

The problem with this approach is that they are then using outdated dependencies. This works for Debian Sid or Arch but not nix.

jerith666 commented:
Well, the JDK itself is an outdated dependency in this scenario, so I'm not sure it's really a new problem if the JDK's dependencies are also outdated.

The JDK in the previous stable NixOS release should see those dependency updates at least for a while, so that would also be an option for users who need it, at least for a short time.

All this being said, I certainly wouldn't reject a PR that gets JDK 16 into nixpkgs only because it also retains JDK 15!  :-)

SuperSandro2000 commented:
@aanderse I think this PR might break kodi.

aanderse commented:
https://forum.kodi.tv/showthread.php?tid=233523&pid=2065257#pid2065257
> java is needed for swig, which we use to generate parts of our code(interfaces like json-rpc and python).
It is only needed during bootstrap phase.
It is possible to ship pre-bootstrapped code, which is what gentoo does iirc. 

This makes me wonder if we even actually need `java` in `kodi`... 🤔

jlesquembre commented:
I was testing the openjdk 16 version from this PR, and it works fine, I didn't find any problems :) thanks @petabyteboy

Commit:
jdk: update darwin to 16 as well

petabyteboy commented:
I renamed the file to 16.nix to make further renames betwen `<version>.nix` and `default.nix` unnecessary.

I also added some options to make the version string more sensible:

```
[pbb@onyx:~/proj/nixpkgs]$ result/bin/java -version
openjdk version "16" 2021-03-16
OpenJDK Runtime Environment (build 16+36-nixos)
OpenJDK 64-Bit Server VM (build 16+36-nixos, mixed mode, sharing)

[pbb@onyx:~/proj/nixpkgs]$
```

I added another commit that updates the binary zulu jdk distribution for darwin, which is available under the same openjdk16 attribute.

SuperSandro2000 commented:
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).
If you have any questions or problems please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 116717` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
26 packages marked as broken and skipped:
</summary>
<ul>
<li>jboss</li>
<li>junixsocket</li>
<li>kodiPackages.inputstream-adaptive</li>
<li>kodiPackages.inputstream-ffmpegdirect</li>
<li>kodiPackages.inputstream-rtmp</li>
<li>kodiPackages.joystick</li>
<li>kodiPackages.pvr-hdhomerun</li>
<li>kodiPackages.pvr-hts</li>
<li>kodiPackages.pvr-iptvsimple</li>
<li>kodiPackages.steam-controller</li>
<li>kodiPackages.vfs-libarchive</li>
<li>kodiPackages.vfs-sftp</li>
<li>mindustry</li>
<li>mindustry-server</li>
<li>mindustry-wayland</li>
<li>octave-jit</li>
<li>odpdown</li>
<li>openfire</li>
<li>picolisp</li>
<li>plm</li>
<li>portmod</li>
<li>python38Packages.skein</li>
<li>python39Packages.skein</li>
<li>shattered-pixel-dungeon</li>
<li>spark</li>
<li>vuze</li>
</ul>
</details>
<details>

  <summary>6 packages failed to build and already failed to build on hydra master:</summary>
  <ul>
<li>graalvm8: log was empty</li>
<li>haxePackages.hxjava: log was empty</li>
<li>librsb: log was empty</li>
<li>octaveFull: log was empty</li>
<li>scenebuilder: log was empty</li>
<li>scenic-view: log was empty</li>
  </ul>
</details>
<details>

  <summary>3 packages failed to build and are new build failures:</summary>
  <ul>
<li>plantuml-server: log was empty</li>
<li>python39Packages.sphinxcontrib_plantuml: log was empty</li>
<li>sourcetrail: log was empty</li>
  </ul>
</details>
<details>
<summary>
121 packages built:
</summary>
<ul>
<li>ammonite (ammonite_2_13)</li>
<li>ammonite_2_12</li>
<li>apktool</li>
<li>asciidoc-full-with-plugins</li>
<li>asciidoctorj</li>
<li>aspectj</li>
<li>astrolabe-generator</li>
<li>awstats</li>
<li>axis2</li>
<li>bfg-repo-cleaner</li>
<li>bftools</li>
<li>bloop</li>
<li>boot</li>
<li>briss</li>
<li>cfr</li>
<li>clooj</li>
<li>closurecompiler</li>
<li>confluent-platform</li>
<li>davmail</li>
<li>ditaa</li>
<li>domination</li>
<li>emem</li>
<li>emscripten</li>
<li>epubcheck</li>
<li>ergo</li>
<li>exhibitor</li>
<li>flink</li>
<li>flyway</li>
<li>frostwire</li>
<li>gama</li>
<li>geoipjava</li>
<li>global-platform-pro</li>
<li>gogui</li>
<li>gpsprune</li>
<li>gremlin-console</li>
<li>groove</li>
<li>groovy</li>
<li>hsqldb</li>
<li>html5validator</li>
<li>hydraAntLogger</li>
<li>imagej</li>
<li>jadx</li>
<li>jameica</li>
<li>jbang</li>
<li>jd-gui</li>
<li>jdiskreport</li>
<li>jedit</li>
<li>jflex</li>
<li>jing (jing-trang)</li>
<li>jmeter</li>
<li>jmusicbot</li>
<li>jruby</li>
<li>jugglinglab</li>
<li>key</li>
<li>kotlin</li>
<li>leiningen</li>
<li>liquibase</li>
<li>logisim</li>
<li>lombok</li>
<li>ma1sd</li>
<li>macse</li>
<li>maven (maven3)</li>
<li>mediathekview</li>
<li>micronaut</li>
<li>mill</li>
<li>minc_widgets</li>
<li>minecraft-server</li>
<li>mkgmap</li>
<li>mkgmap-splitter</li>
<li>msgviewer</li>
<li>mxisd</li>
<li>nexus</li>
<li>octave</li>
<li>opa</li>
<li>papermc</li>
<li>perl530Packages.NetZooKeeper</li>
<li>perl532Packages.NetZooKeeper</li>
<li>picard-tools</li>
<li>plantuml</li>
<li>prometheus-jmx-httpserver</li>
<li>python38Packages.databricks-connect</li>
<li>python38Packages.sphinxcontrib_plantuml</li>
<li>python39Packages.databricks-connect</li>
<li>quantomatic</li>
<li>rascal</li>
<li>riemann</li>
<li>runelite</li>
<li>saxon</li>
<li>saxon-he</li>
<li>saxonb (saxonb_8_8)</li>
<li>saxonb_9_1</li>
<li>sbt (simpleBuildTool)</li>
<li>sbt-extras</li>
<li>sbt-with-scala-native</li>
<li>selendroid</li>
<li>selenium-server-standalone</li>
<li>seyren</li>
<li>snpeff</li>
<li>soapui</li>
<li>solr</li>
<li>sonar-scanner-cli</li>
<li>spring-boot-cli</li>
<li>sshoogr</li>
<li>stm32cubemx</li>
<li>storm</li>
<li>subgit</li>
<li>subsonic</li>
<li>swingsane</li>
<li>tabula</li>
<li>tabula-java</li>
<li>tomcat-native</li>
<li>tomcat_connectors</li>
<li>umlet</li>
<li>varscan</li>
<li>vassal</li>
<li>visualvm</li>
<li>weka</li>
<li>xmloscopy</li>
<li>yuicompressor</li>
<li>zipkin</li>
<li>zookeeper_mt</li>
</ul>
</details>

petabyteboy commented:
Thanks for testing! I don't see the jdk in there, but it must be because those things depend on it.

I would say let's apply this and fix the fallout. In the worst case, some packages will have to be downgraded to jdk11 until there's a new LTS release (I think 17 will be LTS?).
